test/simulator/timestamp/src/connection_simulator.cpp

```diff
@@ -86,78 +86,81 @@ connection_simulator::query_timestamp() const
     return (0);
 }
 
-/*
- * Parse a single timestamp configuration string eg. oldest_timestamp=10. The string is split on the
- * '=' and the new timestamp is assigned.
- */
-int
-connection_simulator::parse_timestamp_config_single(
-  const std::string &config, uint64_t *new_oldest_ts, uint64_t *new_stable_ts)
+/* Parse a single timestamp configuration string eg. oldest_timestamp=10. */
+bool
+connection_simulator::parse_timestamp_config_single(const std::string &config,
+  uint64_t *new_oldest_ts, uint64_t *new_stable_ts, uint64_t *new_durable_ts)
 {
-
-    /* The substring before the '=' indicates the type of timestamp to be set. */
-    const std::string ts_type = config.substr(0, config.find("="));
-    /* Copy the substring after the '=' to get the timestamp value from the config string. */
-    std::string ts_string = config.substr(config.find("=") + 1);
-    /* Convert the timestamp to an int. */
-    uint64_t ts = std::stoi(ts_string);
-
-    switch (system_timestamps_map.at(ts_type)) {
-    case oldest_timestamp:
-        *new_oldest_ts = ts;
-        break;
-    case stable_timestamp:
-        *new_stable_ts = ts;
-        break;
-    case durable_timestamp:
-        _durable_ts = ts;
-        break;
+    int durable_found = config.find("durable_timestamp=");
+    int oldest_found = config.find("oldest_timestamp=");
+    int stable_found = config.find("stable_timestamp=");
+
+    std::string ts_string;
+    if (oldest_found != -1) {
+        ts_string = config.substr(oldest_found + 17);
```

Comment:
I don't really like those magic numbers: `17` and `18`. Can we do instead `oldest_found.length()` or `oldest_found.size()`?

Reply:
Updated in [57ebfa0](https://github.com/wiredtiger/wiredtiger/pull/8116/commits/57ebfa0c661435991016c35a06922cddd530d2b3)